goopicsi.go

```diff
@@ -126,6 +126,50 @@ func NVMeControllerConnect(request *pb.NVMfRemoteController) (*pb.NVMfRemoteCont
 	return &pb.NVMfRemoteControllerConnectResponse{}, nil
 }
 
+// NVMeControllerList lists all the connections to the remote NVMf controller
+func NVMeControllerList() (*pb.NVMfRemoteControllerListResponse, error) {
+	if conn == nil {
+		err := dialConnection()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	client := pb.NewNVMfRemoteControllerServiceClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	response, err := client.NVMfRemoteControllerList(ctx, &pb.NVMfRemoteControllerListRequest{})
+	if err != nil {
+		log.Printf("could not list the connections to Remote NVMf controller: %v", err)
+		return nil, err
+	}
+	log.Printf("Connections: %v", response)
+	return response, nil
+}
+
+// NVMeControllerGet lists the connection to the remote NVMf controller corresponding to the given ID
+func NVMeControllerGet(request *pb.NVMfRemoteControllerGetRequest) (*pb.NVMfRemoteControllerGetResponse, error) {
```

Comment:
Here as well, Avoid using the proto messages directly on the request and response objects.

Reply:
Fixed